customerio/customerio-ios
# refactor: fix uikit inline View issues after SwiftUI architecture changes

Ticket: https://linear.app/customerio/issue/MBL-402/

# Problem 

The [previous PR](https://github.com/customerio/customerio-ios/pull/782) made an architecture change to how animations are performed for displaying inline in-app messages. The previous PR was the SwiftUI implementation, this PR is the UIKit implementation. 

# Solution 

This PR is not adding new logic but instead doing a refactor that splits the existing `InAppMessageView` UIKit View into 2 separate UIKit UIViews. 

1. Create a generic UIKit UIView that renders inline in-app messages and sends events to a delegate. 
2. Create a UIKit wrapper around this generic View that implements features such as: animating height changes and showing a loading View when the "show another message" action buttons are clicked. 

These 2 classes match the proposed architecture changes the last PR mentioned: 

```
SwiftUI View <--- performs animations when height changes
  \__ UIKit WebView renders in-app message

UIKit View <--- performs animations when height changes 
  \__ UIKit WebView renders in-app message
```

# Testing 

I tested this scenario on both sample apps. 

* Open Dashboard in sample app. Send an in-app message. Expect inline View to be height 0 when first displayed. 
* When in-app message renders, animate height to N to display the in-app message. 
* Click action button that displays a new message. Expect see a an Activity Indicator when the button is clicked and next message is rendering. 
* When the next message renders, expect the height to animate to the new height of the inline message. 
* Click close button. Because there is no other message in local queue, animate height to 0 to dismiss in-app message. 

Demo for SwiftUI app:

https://github.com/user-attachments/assets/65eb2f0b-aa95-479a-ae02-e21ab5018893

Demo for UIKit app: 

https://github.com/user-attachments/assets/53c2df20-e294-4266-96b2-ee3f32fe05db

# Follow-up PRs 

There are some changes that are either out-of-scope for this PR or changes that would make this PR large and hard to review. Expect to see these changes in a follow-up PR: 

1. Fix automated tests for inline View. 
2. Modify the public class name for the UIKit and SwiftUI inline Views.
